rfc/resolve-directive.md

```diff
@@ -0,0 +1,189 @@
+---
+author: Jens Neuse
+title: @resolve directive to resolve input fields or entity request inputs in the Router
+---
+
+## Problem
+
+It's common to have a Subgraph that can mutate entities which have references to other entities.
+When a mutation is performed, we've got to ensure that the referenced entity exists or if the ID is valid.
+
+The problem could be solved by making a network call from one Subgraph to another to validate the integrity of the reference.
+However, this approach would introduce multiple problems.
+
+1. We create a dependency between Subgraphs which creates tight coupling
+2. Schema usage metrics only work for requests that are resolved through the Router
+
+We're adding unnecessary complexity and overhead to the system by requiring network calls between Subgraphs.
+We could solve this issue by calling other Subgraphs through the Router,
+but this would require us the expose the validation query to the public API,
+or we'd have to run a separate instance of the Router for internal use with a contract,
+which adds complexity again.
+
+Wouldn't it be great if we could somehow validate references in a declarative way?
+In addition, such a solution could allow us to "query" input fields for operations that are not directly related to the entity.
+
+## Solution
+
+I propose to add "internal" fields which we mark as `@inaccessible`.
+These can be used by the Router to resolve information that is useful to other Subgraphs, but not directly accessible as keys.
+
+In addition, I propose a new directive `@resolve` which allows us to define a query on an input field.
+This input field can be market as `@inaccessible` to prevent exposing it to the public API.
+As such, clients are not able to define the input field in their queries.
+Instead, the Router will execute the query defined in the `@resolve` directive and pass the result to the input.
```

Comment:
`@resolve` only makes sense as a name if we intend this functionality for several use cases.

In this RFC, what is actually being conducted is a prerequisite or a prevalidation.